src/system.rs

```diff
@@ -1023,6 +1023,14 @@ macro_rules! system {
             }
         }
 
+        // Safe because our storage type is Unpin
```

Comment:
I think it would be more helpful to comment on the lack of an `Unpin` bound on `D` and `U` as the `V: Unpin` is there in the code and the lack of a bound is also what differentiates this from the automatically generated `impl` AFAIU.